docs/check_spelling.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+set -e
+
+# This script does spell checks and can add words to a dictionary.
+# It can be used as a git hook to perform checks, it should be named 
+# "pre-push","pre-commit" based on the use case and kept into the ".git/hooks/"
+# and be executable.It can be bypassed by--no-verify option with git command.
+
+# Language of the doc, install appropriate aspell libs when using 
+# a non-English language
+lang=en
+
+project_dict="./.aspell.en.pws"
```

Comment:
Is there a specific reason why the dictionary is a dot-file? I'm thinking for usability reasons it would make sense to just call it `aspell.en.pws` (without the leading dot) or something like that.

Reply:
Done.